api/spec/controllers/spree/api/v1/checkouts_controller_spec.rb

```diff
@@ -232,15 +232,34 @@ def controller.load_order(*)
 
       it 'allow users to reuse a credit card' do
         order.update_column(:state, 'payment')
-        credit_card = create(:credit_card, user_id: order.user_id, payment_method_id: @payment_method.id)
+        credit_card = create(:credit_card, user_id: order.user_id, payment_method_id: @credit_card_payment_method.id)
 
         api_put :update, id: order.to_param, order_token: order.token,
-                         order: { existing_card: credit_card.id }
+                         order: {
+                           existing_card: credit_card.id,
+                           payments_attributes: [{ payment_method_id: @credit_card_payment_method.id }]
+                         }
 
         expect(response.status).to eq 200
         expect(order.credit_cards).to match_array [credit_card]
       end
 
+      it 'allow users to use check as payment method' do
+        check_payment_method = create(:check_payment_method)
+
+        order.update_column(:state, 'payment')
+        credit_card = create(:credit_card, user_id: order.user_id, payment_method_id: @credit_card_payment_method.id)
+
+        api_put :update, id: order.to_param, order_token: order.token,
```

Comment:
IMHO something like this would be more readable
```suggestion
        api_put(
          	:update,
          	id: order.to_param,
          	order_token: order.token,
            order: {
           	  existing_card: credit_card.id,
              payments_attributes: [{ payment_method_id: check_payment_method.id }]
            })
```